src/directives/schema-validate.js

```diff
@@ -1,110 +1,190 @@
-        // Check if our version of angular has i, i.e. 1.3+
-        if (form[attr] && ngModel[attr]) {
-          angular.forEach(form[attr], function(fn, name) {
-            ngModel[attr][name] = fn;
-          });
-        }
-      });
-
-      // Get in last of the parses so the parsed value has the correct type.
-      // We don't use $validators since we like to set different errors depeding tv4 error codes
-      ngModel.$parsers.push(validate);
-
-      // Listen to an event so we can validate the input on request
-      scope.$on('schemaFormValidate', function() {
-        if (ngModel.$setDirty) {
-          // Angular 1.3+
-          ngModel.$setDirty();
-          validate(ngModel.$modelValue);
-        } else {
-          // Angular 1.2
-          ngModel.$setViewValue(ngModel.$viewValue);
+        // Get in last of the parses so the parsed value has the correct type.
+        // We don't use $validators since we like to set different errors depeding tv4 error codes
+        ngModel.$parsers.push(validate);
+
+        // Listen to an event so we can validate the input on request
+        scope.$on('schemaFormValidate', function() {
+          if (ngModel.$setDirty) {
+            // Angular 1.3+
+            ngModel.$setDirty();
+            validate(ngModel.$modelValue);
+          } else {
+            // Angular 1.2
+            ngModel.$setViewValue(ngModel.$viewValue);
+          }
+
+        });
+
+
+        var DEFAULT_DESTROY_STRATEGY = getGlobalOptionsDestroyStrategy();
+
+        function getGlobalOptionsDestroyStrategy() {
+          var defaultStrategy = undefined;
+          if (scope.options && scope.options.hasOwnProperty('destroyStrategy')) {
+            var globalOptionsDestroyStrategy = scope.options.destroyStrategy;
+            var isValidFormDefaultDestroyStrategy = (globalOptionsDestroyStrategy === undefined ||
+                                                    globalOptionsDestroyStrategy === '' ||
+                                                    globalOptionsDestroyStrategy === null ||
+                                                    globalOptionsDestroyStrategy === 'retain');
+            if (isValidFormDefaultDestroyStrategy) {
+              defaultStrategy = globalOptionsDestroyStrategy;
+            }
+            else {
+              console.warn('Unrecognized globalOptions destroyStrategy: %s \'%s\'. Used undefined instead.',
+                  typeof globalOptionsDestroyStrategy, globalOptionsDestroyStrategy);
+            }
+          }
+          return defaultStrategy;
         }
 
-      });
+        // Clean up the model when the corresponding form field is $destroy-ed.
+        // Default behavior can be supplied as a globalOption, and behavior can be overridden in the form definition.
+        scope.$on('$destroy', function() {
+
+          var form = getForm();
+          var conditionResult = $parse(form.condition);
```

Comment:
I had left it in as a way to help show where I was coming from with my solution, because I wasn't 100% confident that the service I put in would be correct (I'd still label myself an apprentice with Javascript). Assuming that the flag will handle for the case where the destroy is coming from outside of the form, then I agree that destroys initiated within the form should just follow the destroyStrategy. It'll be more consistent (and still configurable) and easier to explain.